tutorials/intro/plot_introduction.py

```diff
@@ -1,404 +1,368 @@
+# results of the rejection criteria being applied:
+
+epochs = mne.Epochs(raw, events, event_id=event_dict, tmin=-0.2, tmax=0.5,
+                    reject=reject_criteria, preload=True)
+
+###############################################################################
+# Next we'll pool across left/right stimulus presentations so we can compare
+# auditory versus visual responses. To avoid biasing our signals to the
+# left or right, we'll use :meth:`~mne.Epochs.equalize_event_counts` first to
+# randomly sample epochs from each condition to match the number of epochs
+# present in the condition with the fewest good epochs.
+
+conds_we_care_about = ['auditory/left', 'auditory/right',
+                       'visual/left', 'visual/right']
+epochs.equalize_event_counts(conds_we_care_about)  # this operates in-place
+aud_epochs = epochs['auditory']
+vis_epochs = epochs['visual']
+del raw, epochs  # free up memory
+
+###############################################################################
+# Like :class:`~mne.io.Raw` objects, :class:`~mne.Epochs` objects also have a
+# number of built-in plotting methods. One is :meth:`~mne.Epochs.plot_image`,
+# which shows each epoch as one row of an image map, with color representing
+# signal magnitude; the average evoked response and the sensor location are
+# shown below the image:
+
+aud_epochs.plot_image(picks=['MEG 1332', 'EEG 021'])
+
+##############################################################################
+# .. note::
 #
-# Scipy supports read and write of matlab files. You can save your single
-# trials with:
-
-from scipy import io  # noqa
-io.savemat('epochs_data.mat', dict(epochs_data=epochs_data), oned_as='row')
-
-##############################################################################
-# or if you want to keep all the information about the data you can save your
-# epochs in a fif file:
-
-epochs.save('sample-epo.fif', overwrite=True)
-
-##############################################################################
-# and read them later with:
-
-saved_epochs = mne.read_epochs('sample-epo.fif')
-
-##############################################################################
-# Compute evoked responses for auditory responses by averaging and plot it:
-
-evoked = epochs['aud_l'].average()
-print(evoked)
-evoked.plot(time_unit='s')
-
-##############################################################################
-# .. topic:: Exercise
+#     Both :class:`~mne.io.Raw` and :class:`~mne.Epochs` objects have
+#     :meth:`~mne.Epochs.get_data` methods that return the underlying data
+#     as a :class:`NumPy array <numpy.ndarray>`. Both methods have a ``picks``
+#     parameter for subselecting which channel(s) to return; ``raw.get_data()``
+#     has additional parameters for restricting the time domain. The resulting
+#     matrices have dimension ``(n_channels, n_times)`` for
+#     :class:`~mne.io.Raw` and ``(n_epochs, n_channels, n_times)`` for
+#     :class:`~mne.Epochs`.
```

Comment:
I prefer the "show and tell" approach for the data shape. Any reason you chose to have a note instead of printing ``raw.get_data().shape`` and then ``raw.info['nchan']``

Reply:
in my mind, it makes more sense as a note because you don't actually need to `get_data()` in order to achieve the analysis results of this tutorial.  It's more of a "in case you're wondering..." comment.  It will of course be demonstrated more thoroughly in other tutorials (like [this one](https://12627-1301584-gh.circle-artifacts.com/0/dev/auto_tutorials/continuous_data/15_subselecting_raw_data.html#sphx-glr-auto-tutorials-continuous-data-15-subselecting-raw-data-py)).